Translate /docs/concepts/cluster-administration/cloud-providers.md into Chinese (#17761)
kubernetes/website: content/zh/docs/concepts/cluster-administration/cloud-providers.md

---

> title: 云供应商

Reviewer: 在这个上下文中，cloud provider 的 provider 本质上是一个 driver，用来提供 Kubernetes 下层的基础设施。
不适合翻译为“供应商”。

    mountPath: "/etc/kubernetes/cloud.conf"
controllerManager:
  extraArgs:
    cloud-provider: "openstack"
    cloud-config: "/etc/kubernetes/cloud.conf"
  extraVolumes:
  - name: cloud
    hostPath: "/etc/kubernetes/cloud.conf"
    mountPath: "/etc/kubernetes/cloud.conf"
```

<!--
The in-tree cloud providers typically need both `--cloud-provider` and `--cloud-config` specified in the command lines
for the [kube-apiserver](/docs/admin/kube-apiserver/), [kube-controller-manager](/docs/admin/kube-controller-manager/) and the
[kubelet](/docs/admin/kubelet/). The contents of the file specified in `--cloud-config` for each provider is documented below as well.

For all external cloud providers, please follow the instructions on the individual repositories,
which are listed under their headings below, or one may view [the list of all repositories](https://github.com/kubernetes?q=cloud-provider-&type=&language=)
-->

“树内”的云供应商通常需要在命令行中为 [kube-apiserver](/docs/admin/kube-apiserver/)、[kube-controller-manager](/docs/admin/kube-controller-manager/) 和 [kubelet](/docs/admin/kubelet/) 指定 “--cloud-provider” 和 “--cloud-config”。在 “--cloud-config” 中为每个供应商指定的文件的内容也同样需要写在下面。

> 对于所有外部云供应商，请遵循独立云存储库的说明，或浏览[所有版本库清单]（https://github.com/kubernetes?q=cloud-provider-&type=&language=）

Reviewer: `[anchor](link)` 形式的语法是 markdown 中用来标示超级链接的，其中的中括号和圆括号都必须是半角。

<!--
## AWS
This section describes all the possible configurations which can
be used when running Kubernetes on Amazon Web Services.

If you wish to use the external cloud provider, its repository is [kubernetes/cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws#readme)
-->

# AWS
本节介绍在 Amazon Web Services 上运行 Kubernetes 时可以使用的所有配置。

> 如果希望使用外部云供应商，可以使用它的版本库 [kubernetes/cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws#readme)

Reviewer: ```suggestion
如果希望此外部云驱动，其代码库位于 [kubernetes/cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws#readme)。
```

<!--
### Node Name

The AWS cloud provider uses the private DNS name of the AWS instance as the name of the Kubernetes Node object.
-->
### 节点名称

> 云供应商AWS使用AWS实例的私有 DNS 名称作为 Kubernetes 节点对象的名称。

Reviewer: 注意中英文之间留白。

  selector:
    app: example
```
<!--
Different settings can be applied to a load balancer service in AWS using _annotations_. The following describes the annotations supported on AWS ELBs:
-->

> 可以使用 _注解_ 将不同的设置应用于 AWS 中的负载平衡器服务。下面描述了 AWS ELB 所支持的注解：

Reviewer: loadbalance -> 负载均衡

* `service.beta.kubernetes.io/aws-load-balancer-security-groups`: Used to specify the security groups to be added to ELB created. This replaces all other security groups previously assigned to the ELB.
* `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups`: Used on the service to specify additional security groups to be added to ELB created
* `service.beta.kubernetes.io/aws-load-balancer-internal`: Used on the service to indicate that we want an internal ELB.
* `service.beta.kubernetes.io/aws-load-balancer-proxy-protocol`: Used on the service to enable the proxy protocol on an ELB. Right now we only accept the value `*` which means enabling the proxy protocol on all ELB backends. In the future we could adjust this to allow setting the proxy protocol only on certain backends.
* `service.beta.kubernetes.io/aws-load-balancer-ssl-ports`: Used on the service to specify a comma-separated list of ports that will use SSL/HTTPS listeners. Defaults to `*` (all)
-->

* `service.beta.kubernetes.io/aws-load-balancer-access-log-emit-interval`：用于指定访问日志的间隔。

> * `service.beta.kubernetes.io/aws-load-balancer-access-log-enabled`：用于在服务中启用或禁用日志访问。

Reviewer: access log 是访问日志，众多日志当中的一种，不能译为“日志访问”

* `service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout`：用于在服务中指定空闲连接超时时间。
* `service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled`：用于在服务中启用或禁用跨区域负载平衡。
* `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups`：用于在服务中指定要添加到创建的 ELB 中的其他安全组。
* `service.beta.kubernetes.io/aws-load-balancer-internal`：用于在服务中表明需要内部 ELB。
* `service.beta.kubernetes.io/aws-load-balancer-proxy-protocol`：用于在 ELB 上启用代理协议。 当前仅接受 `*`值，也就是在所有 ELB 后端启用代理协议。将来可能进行调整，只允许特定的后端设置代理协议。
* `service.beta.kubernetes.io/aws-load-balancer-ssl-ports`：用于在服务中指定一个逗号分隔的端口列表，这些端口会使用 SSL/HTTPS 监听器。默认为 `*`（全部）

<!--
The information for the annotations for AWS is taken from the comments on [aws.go](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/cloudprovider/providers/aws/aws.go)
-->

AWS 相关的注解信息取自 [aws.go](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/cloudprovider/providers/aws/aws.go) 文件的注释。

## Azure

<!--
If you wish to use the external cloud provider, its repository is [kubernetes/cloud-provider-azure](https://github.com/kubernetes/cloud-provider-azure#readme)
-->

> 如果希望使用外部云供应商，可以使用它的版本库 [kubernetes/cloud-provider-azure](https://github.com/kubernetes/cloud-provider-azure#readme)

Reviewer: 同前

<!--
### Node Name

The Azure cloud provider uses the hostname of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
Note that the Kubernetes Node name must match the Azure VM name.
-->

### 节点名称

云供应商 Azure 使用节点的主机名（由 kubelet 决定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。
注意 Kubernetes 节点名必须与 Azure 虚拟机的名称匹配。

## CloudStack

<!--
If you wish to use the external cloud provider, its repository is [apache/cloudstack-kubernetes-provider](https://github.com/apache/cloudstack-kubernetes-provider)
-->

> 如果希望使用外部云供应商，可以使用它的版本库 [apache/cloudstack-kubernetes-provider](https://github.com/apache/cloudstack-kubernetes-provider)

Reviewer: 同前

<!--
### Node Name

The CloudStack cloud provider uses the hostname of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
Note that the Kubernetes Node name must match the CloudStack VM name.
-->

### 节点名称

云供应商 CloudStack 使用节点的主机名（由 kubelet 决定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。
注意 Kubernetes 节点名必须与 CloudStack 虚拟机名匹配。

## GCE
<!--
If you wish to use the external cloud provider, its repository is [kubernetes/cloud-provider-gcp](https://github.com/kubernetes/cloud-provider-gcp#readme)
-->

> 如果希望使用外部云供应商，可以使用它的版本库 [kubernetes/cloud-provider-gcp](https://github.com/kubernetes/cloud-provider-gcp#readme)

Reviewer: 同前

<!--
### Node Name

The GCE cloud provider uses the hostname of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
Note that the first segment of the Kubernetes Node name must match the GCE instance name (e.g. a Node named `kubernetes-node-2.c.my-proj.internal` must correspond to an instance named `kubernetes-node-2`).
-->

### 节点名称

> 云供应商 GCE 使用节点的主机名（由 kubelet 确定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。

Reviewer: ```suggestion
GCE 云驱动使用节点的主机名（由 kubelet 确定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。
```

> 注意，Kubernetes 节点名的第一个字段必须匹配 GCE 实例名(例如，名为 “Kubernetes - Node -2.c.my-proj.internal” 的节点必须对应于一个名为 “Kubernetes - Node -2” 的实例)。

Reviewer: 去掉 “Kubernetes - Node -2” 中间的空格

## OpenStack
<!--
This section describes all the possible configurations which can
be used when using OpenStack with Kubernetes.

If you wish to use the external cloud provider, its repository is [kubernetes/cloud-provider-openstack](https://github.com/kubernetes/cloud-provider-openstack#readme)
-->

本节介绍了使用 OpenStack 运行 Kubernetes 时所有可用的配置。

> 如果希望使用外部云供应商，可以使用它的版本库 [kubernetes/cloud-provider-openstack](https://github.com/kubernetes/cloud-provider-openstack#readme)

Reviewer: 同前

<!--
### Node Name

The OpenStack cloud provider uses the instance name (as determined from OpenStack metadata) as the name of the Kubernetes Node object.
Note that the instance name must be a valid Kubernetes Node name in order for the kubelet to successfully register its Node object.
-->

### 节点名称

OpenStack 云供应商使用实例名（由 OpenStack 元数据确定）作为 Kubernetes 节点对象的名称。
请注意，实例名必须是一个有效的 Kubernetes 节点名，以便 kubelet 成功注册其节点对象。

<!--
### Services

The OpenStack cloud provider
implementation for Kubernetes supports the use of these OpenStack services from
the underlying cloud, where available:

| Service                  | API Version(s) | Required |
|--------------------------|----------------|----------|
| Block Storage (Cinder)   | V1†, V2, V3    | No       |
| Compute (Nova)           | V2             | No       |
| Identity (Keystone)      | V2‡,  V3       | Yes      |
| Load Balancing (Neutron) | V1§, V2        | No       |
| Load Balancing (Octavia) | V2             | No       |

-->
### 服务

Kubernetes 的 OpenStack 云供应商实现支持从底层云使用这些 OpenStack 服务。

> 为 Kubernetes 实施的 OpenStack 支持从云的底层使用这些 OpenStack 服务：

Reviewer: 277 行无中生有

| 服务                     | API 版本       | 必须     |
|--------------------------|----------------|----------|
| 块存储 (Cinder)          | V1†, V2, V3    | No       |
| 计算    (Nova)           | V2             | No       |
| 身份认证 (Keystone)      | V2‡,  V3       | Yes      |
| 负载均衡器 (Neutron)     | V1§, V2        | No       |
| 负载均衡器 (Octavia)     | V2             | No       |

<!--
† Block Storage V1 API support is deprecated, Block Storage V3 API support was
added in Kubernetes 1.9.

‡ Identity V2 API support is deprecated and will be removed from the provider in
a future release. As of the "Queens" release, OpenStack will no longer expose the
Identity V2 API.

§ Load Balancing V1 API support was removed in Kubernetes 1.9.
-->

> † 块存储 V1 版本的 API 被弃用，从 Kubernetes 1.9 版本开始加入了块存储 V3 版本 API。

Reviewer: Block Storage 是 API 的名称，不要翻译，下面的 Identity、Load Balancing 都不要翻译

‡ 身份认证 V2 API 支持已被弃用，将在未来的版本中从供应商中移除。从 “Queens” 版本开始，OpenStack 将不再支持身份认证 V2 版本的 API。

§ Kubernetes 1.9 中取消了对 V1 版本负载均衡API的支持。

<!--
Service discovery is achieved by listing the service catalog managed by
OpenStack Identity (Keystone) using the `auth-url` provided in the provider
configuration. The provider will gracefully degrade in functionality when
OpenStack services other than Keystone are not available and simply disclaim
support for impacted features. Certain features are also enabled or disabled
based on the list of extensions published by Neutron in the underlying cloud.
-->

服务发现是通过使用供应商配置中提供的 “auth-url” 所列出 OpenStack 身份认证 （Keystone）管理的服务目录来实现的。当除 Keystone 外的 OpenStack 服务不可用时，供应商将优雅地降低功能，并简单地放弃对受影响特性的支持。某些功能还可以根据 Neutron 在底层云中发布的扩展列表启用或禁用。

### cloud.conf
<!--
Kubernetes knows how to interact with OpenStack via the file cloud.conf. It is
the file that will provide Kubernetes with credentials and location for the OpenStack auth endpoint.
You can create a cloud.conf file by specifying the following details in it
-->

Kubernetes 知道如何通过 cloud.conf 文件与 OpenStack 交互。该文件将为 Kubernetes 提供 OpenStack 验证端点的凭据和位置。

> 用户可以通过在 cloud.conf 文件中指定以下信息来创建它：

Reviewer: ```suggestion
用户可在创建 cloud.conf 文件时指定以下信息：
```

<!--
#### Typical configuration
This is an example of a typical configuration that touches the values that most
often need to be set. It points the provider at the OpenStack cloud's Keystone
endpoint, provides details for how to authenticate with it, and configures the
load balancer:
-->
#### 典型配置

> 这是一个典型配置的例子，它涉及到最常设置的值。它将供应商指向 OpenStack 云的关键端点，提供如何使用它进行身份验证的细节，并配置负载均衡器:

Reviewer: keystone endpoint 不是关键端点

[LoadBalancer]
subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a
```
<!--
##### Global
These configuration options for the OpenStack provider pertain to its global
configuration and should appear in the `[Global]` section of the `cloud.conf`
file:
-->

##### 全局配置

> 这些配置选项属于 OpenStack 提供程序的全局配置，并且应该出现在 “cloud.conf” 文件中的 “[global]” 部分:

Reviewer: ```suggestion
这些配置选项属于 OpenStack 提供程序的全局配置，并且应该出现在 `cloud.conf` 文件中的 `[global]` 部分:
```

反引号是用来标记代码或软件名称的，不要随意更换。

  an OpenStack deployment. Although a region does not have a strict geographical
  connotation, a deployment can use a geographical name for a region identifier
  such as `us-east`. Available regions are found under the `/v3/regions`
  endpoint of the Keystone API.
* `ca-file` (Optional): Used to specify the path to your custom CA file.
-->

* `auth-url` (必须): 用于认证的 keystone API 的 URL。在 OpenStack 控制面板中，这可以在“访问和安全（Access and Security）> API 访问（API Access）> 凭证（Credentials）”中找到。
* `username` (必须): 指 keystone 中一个有效用户的用户名。
* `password` (必须): 指 keystone 中一个有效用户的密码。
* `tenant-id` (必须): 用于指定要创建资源的租户 ID。
* `tenant-name` (可选): 用于指定要在其中创建资源的租户的名称。
* `trust-id` (可选): 用于指定用于授权的信任的标识符。信任表示用户（委托人）将角色委托给另一个用户(受托人)的授权，并可选的允许受托人模仿委托人。可用的信任可以在 Keystone API 的 “/v3/OS-TRUST/trusts” 端点下找到。
* `domain-id` (可选): 用于指定用户所属域的 ID。
* `domain-name` (可选): 用于指定用户所属域的名称。
* `region` (可选): 用于指定在多区域 OpenStack 云上运行时使用的区域标识符。区域是 OpenStack 部署的一般划分。虽然区域没有严格的地理含义，但部署可以使用地理名称表示区域标识符，如 “us-east”。可用区域位于 Keystone API 的 “/v3/regions” 端点之下。
* `ca-file` (可选): 用于指定自定义 CA 文件的路径。

<!--
When using Keystone V3 - which changes tenant to project - the `tenant-id` value
is automatically mapped to the project construct in the API.
-->

> 当使用 Keystone V3 时(它将租户更改为项目)，“tenant-id” 值会自动映射到 API 中的项目。

Reviewer: tenant 和 project 不要翻译

<!--
#####  Load Balancer
These configuration options for the OpenStack provider pertain to the load
balancer and should appear in the `[LoadBalancer]` section of the `cloud.conf`
file:
-->

####  负载均衡器

> 这些配置选项属于 OpenStack 提供程序的全局配置，并且应该出现在 “cloud.conf” 文件中的 “[LoadBalancer]” 部分:

Reviewer: 建议同上

  The default is `false`. When `true` is specified then `monitor-delay`,
  `monitor-timeout`, and `monitor-max-retries` must also be set.
* `monitor-delay` (Optional): The time between sending probes to
  members of the load balancer. Ensure that you specify a valid time unit. The valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h"
* `monitor-timeout` (Optional): Maximum time for a monitor to wait
  for a ping reply before it times out. The value must be less than the delay
  value. Ensure that you specify a valid time unit. The valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h"
* `monitor-max-retries` (Optional): Number of permissible ping failures before
  changing the load balancer member's status to INACTIVE. Must be a number
  between 1 and 10.
* `manage-security-groups` (Optional): Determines whether or not the load
  balancer should automatically manage the security group rules. Valid values
  are `true` and `false`. The default is `false`. When `true` is specified
  `node-security-group` must also be supplied.
* `node-security-group` (Optional): ID of the security group to manage.
-->

> * `lb-version` (可选): 用于覆盖自动版本检测。有效值为“v1”或“v2”。如果没有提供值，则自动检测将选择底层 OpenStack 云公开的受支持程度最高的版本。

Reviewer: ```suggestion
* `lb-version` (可选): 用于覆盖自动版本检测。有效值为“v1”或“v2”。如果没有提供值，则自动选择底层 OpenStack 云所支持的最高版本。
```

* `use-octavia` (可选): 用于确定是否查找和使用 Octavia LBaaS V2 服务目录端点。有效值是“true”或“false”。
如果指定了“true”，并且无法找到 Octaiva LBaaS V2 入口，则提供者将退回并尝试寻找一个 Neutron LBaaS V2 端点。默认值是“false”。
* `subnet-id` (可选): 用于指定要在其上创建负载均衡器的子网的 ID。
可以在 “Network > Networks” 上找到。
单击相应的网络以获得其子网。
* `floating-network-id` (可选): 如果指定，将为负载均衡器创建一个浮动 IP。

> * `lb-method` (可选): 用于指定将负载分配到负载均衡器池成员的算法。值可以是' ROUND_ROBIN '、' LEAST_CONNECTIONS '或' SOURCE_IP '。如果没有指定，默认行为是' ROUND_ROBIN '。

Reviewer: 请修改 463 行的空白、引号

* `lb-provider` (可选): 用于指定负载均衡器的提供程序。如果没有指定，将使用在 neutron 中配置的默认提供者服务。
* `create-monitor` (可选): 指定是否为 Neutron 负载均衡器创建健康监视器。有效值是“true”和“false”。
默认为“false”。当指定“true”时，还必须设置 “monitor-delay”、“monitor-timeout” 和 “monitor-max-retries”。
* `monitor-delay` (可选): 向负载均衡器的成员发送探测器之间的时间间隔。
确保您指定了一个有效的时间单位。

> 有效时间单位为 “ns”， “us” (或 “µs”)， “ms”， “s”， “m”， “h”。

Reviewer: 中文排比用顿号，不用逗号

> * `monitor-timeout` (可选): 在超时之前，监视器等待 ping 响应的最长时间。该值必须小于延迟值。确保您指定了一个有效的时间单位。有效时间单位为 “ns”， “us” (或 “µs”)， “ms”， “s”， “m”， “h”。

Reviewer: 改顿号

* `search-order` (Optional): This configuration key influences the way that the
  provider retrieves metadata relating to the instance(s) in which it runs. The
  default value of `configDrive,metadataService` results in the provider
  retrieving metadata relating to the instance from the config drive first if
  available and then the metadata service. Alternative values are:
  * `configDrive` - Only retrieve instance metadata from the configuration
    drive.
  * `metadataService` - Only retrieve instance metadata from the metadata
    service.
  * `metadataService,configDrive` - Retrieve instance metadata from the metadata
    service first if available, then the configuration drive.

  Influencing this behavior may be desirable as the metadata on the
  configuration drive may grow stale over time, whereas the metadata service
  always provides the most up to date view. Not all OpenStack clouds provide
  both configuration drive and metadata service though and only one or the other
  may be available which is why the default is to check both.
-->

##### 元数据

这些配置选项属于 OpenStack 提供程序的全局配置，并且应该出现在 “cloud.conf” 文件中的 “[Metadata]” 部分：

> * `search-order` (可选): 此配置键影响提供者检索与其运行的实例相关的元数据的方式。

Reviewer: 567 行和后面几行是同一个列表，请保持相同的缩进

Author: 貌似不应该是同一缩进。。。

“configDrive，metadataService” 的默认值导致供应商首先从配置驱动器中检索与实例相关的元数据（如果可用的话），然后检索元数据服务。
他们的替代值：
  * `configDrive` - 仅从配置驱动器检索实例元数据。
  * `metadataService` - 仅从元数据服务检索实例元数据。
  * `metadataService,configDrive` - 如果可用，首先从元数据服务检索实例元数据，然后从配置驱动器检索。

影响这种行为可能是可取的，因为配置驱动器上的元数据可能会随着时间的推移而变得陈旧，而元数据服务总是提供最新的数据视图。并不是所有的 OpenStack 云都同时提供配置驱动和元数据服务，可能只有一个或另一个可用，这就是为什么默认情况下要同时检查两个。

<!--
##### Route

These configuration options for the OpenStack provider pertain to the [kubenet]
Kubernetes network plugin and should appear in the `[Route]` section of the
`cloud.conf` file:

* `router-id` (Optional): If the underlying cloud's Neutron deployment supports
  the `extraroutes` extension then use `router-id` to specify a router to add
  routes to.  The router chosen must span the private networks containing your
  cluster nodes (typically there is only one node network, and this value should be
  the default router for the node network).  This value is required to use [kubenet]
  on OpenStack.

[kubenet]: /docs/concepts/cluster-administration/network-plugins/#kubenet
-->

##### 路由

> 这些配置选项属于 OpenStack 提供程序 [kubenet] 的 Kubernetes 网络插件，并且应该出现在 “cloud.conf” 文件中的 “[Route]” 部分:

Reviewer: ```suggestion
这些配置选项属于 OpenStack 驱动为 Kubernetes 网络插件 [kubenet] 提供的设置，并且应该出现在 “cloud.conf” 文件中的 “[Route]” 部分:
```

* `router-id` (可选)：如果底层云的 Neutron 部署支持 “extraroutes” 扩展，则使用 “router-id” 指定要添加路由的路由器。选择的路由器必须跨越包含集群节点的私有网络（通常只有一个节点网络，这个值应该是节点网络的默认路由器）。在 OpenStack 上使用 [kubenet] 时需要这个值。

[kubenet]: /docs/concepts/cluster-administration/network-plugins/#kubenet

{{% /capture %}}

## OVirt

<!--
### Node Name

The OVirt cloud provider uses the hostname of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
Note that the Kubernetes Node name must match the VM FQDN (reported by OVirt under `<vm><guest_info><fqdn>...</fqdn></guest_info></vm>`)
-->

### 节点名称

> OVirt 云供应商使用节点的主机名（由 kubelet 确定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。

Reviewer: 同前

> 注意 Kubernetes 节点名必须与 VM FQDN 匹配（OVirt 在“<vm><guest_info><fqdn>...</fqdn></guest_info></vm>”中说明）。

Reviewer: 这里的反引号必须改回来，否则会被解析成 HTML tag。

## Photon

<!--
### Node Name

The Photon cloud provider uses the hostname of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
Note that the Kubernetes Node name must match the Photon VM name (or if `overrideIP` is set to true in the `--cloud-config`, the Kubernetes Node name must match the Photon VM IP address).
-->
### 节点名称

> Photon 供应商使用节点的主机名(由 kubelet 决定，或者用 “--hostname-override” 覆盖)作为 Kubernetes 节点对象的名称。

Reviewer: 同前

注意，Kubernetes 节点名必须与 Photon VM名匹配（或者，如果在 “--cloud-config” 中将“overrideIP”设置为 true，则 Kubernetes 节点名必须与 Photon VM IP 地址匹配）。

## VSphere

<!--
### Node Name

The VSphere cloud provider uses the detected hostname of the node (as determined by the kubelet) as the name of the Kubernetes Node object.

The `--hostname-override` parameter is ignored by the VSphere cloud provider.
-->
### 节点名称

> VSphere 云供应商使用节点检测到的主机名(由 kubelet 确定)作为 Kubernetes 节点对象的名称。

Reviewer: 同前

VSphere 云供应商会忽略 “--hostname-override” 参数。

## IBM Cloud Kubernetes Service

<!--
### Compute nodes
By using the IBM Cloud Kubernetes Service provider, you can create clusters with a mixture of virtual and physical (bare metal) nodes in a single zone or across multiple zones in a region. For more information, see [Planning your cluster and worker node setup](https://cloud.ibm.com/docs/containers?topic=containers-plan_clusters#plan_clusters).

The name of the Kubernetes Node object is the private IP address of the IBM Cloud Kubernetes Service worker node instance.
-->
### 计算节点

> 通过使用 IBM Cloud Kubernetes 服务供应商，您可以在单个区域或跨区域的多个区域创建虚拟和物理(裸金属)节点的集群。

Reviewer: ```suggestion
通过使用 IBM Cloud Kubernetes Service 驱动，您可以在单个区域或跨区域的多个区（Region）中创建虚拟和物理（裸金属）节点的集群。
```

有关更多信息，请参见[规划您的集群和工作节点设置](https://cloud.ibm.com/docs/containers?topic=containers-plan_clusters#plan_clusters)。

> Kubernetes 节点对象的名称是 IBM Cloud Kubernetes 服务工作节点实例的私有IP地址。

Reviewer: ```suggestion
Kubernetes 节点对象的名称是 IBM Cloud Kubernetes Services 工作节点实例的私有IP地址。
```

<!--
### Networking
The IBM Cloud Kubernetes Service provider provides VLANs for quality network performance and network isolation for nodes. You can set up custom firewalls and Calico network policies to add an extra layer of security for your cluster, or connect your cluster to your on-prem data center via VPN. For more information, see [Planning in-cluster and private networking](https://cloud.ibm.com/docs/containers?topic=containers-cs_network_cluster#cs_network_cluster).

To expose apps to the public or within the cluster, you can leverage NodePort, LoadBalancer, or Ingress services. You can also customize the Ingress application load balancer with annotations. For more information, see [Planning to expose your apps with external networking](https://cloud.ibm.com/docs/containers?topic=containers-cs_network_planning#cs_network_planning).
-->
### 网络

> IBM Cloud Kubernetes 服务供应商提供 vlan，用于提供高质量的网络性能和节点的网络隔离。您可以设置自定义防火墙和 Calico 网络策略来为您的集群添加额外的安全层，或者通过VPN将您的集群连接到自有数据中心。有关更多信息，请参见[规划集群内和私有网络]（https://cloud.ibm.com/docs/containers?topic=containers-cs_network_cluster#cs_network_cluster）。

Reviewer: ```suggestion
IBM Cloud Kubernetes Services 驱动提供 VLAN，用于提供高质量的网络性能和节点间的网络隔离。您可以设置自定义防火墙和 Calico 网络策略来为您的集群添加额外的安全层，或者通过 VPN 将您的集群连接到自有数据中心。有关更多信息，请参见[规划集群内和私有网络](https://cloud.ibm.com/docs/containers?topic=containers-cs_network_cluster#cs_network_cluster)。
```

> 要向公众或集群内部公开应用程序，您可以利用 NodePort、LoadBalancer 或 Ingress 服务。您还可以使用注释自定义 Ingress 应用程序负载均衡器。有关更多信息，请参见[计划使用外部网络公开您的应用程序]（https://cloud.ibm.com/docs/containers?topic=containers-cs_network_planning#cs_network_planning）。

Reviewer: ```suggestion
要向公众或集群内部公开应用程序，您可以利用 NodePort、LoadBalancer 或 Ingress 服务。您还可以使用注释自定义 Ingress 应用程序负载均衡器。有关更多信息，请参见[计划使用外部网络公开您的应用程序](https://cloud.ibm.com/docs/containers?topic=containers-cs_network_planning#cs_network_planning)。
```

<!--
### Storage
The IBM Cloud Kubernetes Service provider leverages Kubernetes-native persistent volumes to enable users to mount file, block, and cloud object storage to their apps. You can also use database-as-a-service and third-party add-ons for persistent storage of your data. For more information, see [Planning highly available persistent storage](https://cloud.ibm.com/docs/containers?topic=containers-storage_planning#storage_planning).
-->

### 存储

> IBM Cloud Kubernetes 服务供应商利用 Kubernetes 本机持久卷，使用户能够将文件、块和云对象存储装载到他们的应用程序中。还可以使用 database-as-a-service 和第三方附加组件来持久存储数据。有关更多信息，请参见[规划高可用性持久存储]（https://cloud.ibm.com/docs/containers?topic=containers-storage_planning#storage_planning）。

Reviewer: ```suggestion
IBM Cloud Kubernetes Services 驱动利用 Kubernetes 原生的持久卷，使用户能够将文件、块和云对象存储装载到他们的应用程序中。还可以使用 database-as-a-service 和第三方附加组件来持久存储数据。有关更多信息，请参见[规划高可用性持久存储](https://cloud.ibm.com/docs/containers?topic=containers-storage_planning#storage_planning)。
```

<!--
## Baidu Cloud Container Engine

### Node Name

The Baidu cloud provider uses the private IP address of the node (as determined by the kubelet or overridden with `--hostname-override`) as the name of the Kubernetes Node object.
Note that the Kubernetes Node name must match the Baidu VM private IP.
-->
## 百度云容器引擎

### 节点名称

> 百度云供应商使用节点的私有 IP 地址（由 kubelet 确定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。

Reviewer: ```suggestion
Baidu 云驱动使用节点的私有 IP 地址（由 kubelet 确定，或者用 “--hostname-override” 覆盖）作为 Kubernetes 节点对象的名称。
```

Author: 谢谢大神，如此详细！！
